app/workers/zync_worker.rb

```diff
@@ -233,16 +234,44 @@ def valid?(event_id)
     true
   end
 
-  def publish_dependencies_events(event_id)
+  def sync_dependencies(event_id)
+    dependency_events = create_dependency_events(event_id)
+
+    publish_dependency_events(dependency_events)
+    enqueue_dependency_batch(event_id, dependency_events)
+  end
+
+  def create_dependency_events(event_id)
     event = EventStore::Repository.find_event!(event_id)
+    event.dependencies.map { |dependency| ZyncEvent.create(event, dependency, metadata: { skip_background_sync: true }) }
```

Comment:
`event.dependencies` being `event` a `ZyncEvent`, and then map around them to create them all, and it is only used from here:
![image](https://user-images.githubusercontent.com/11318903/57846773-3a380500-77d5-11e9-9cc3-4204818ea3e1.png)

Then I am not sure if in `ZyncEvent` we should have a `create_dependencies` containing this logic (this line) directly instead of return the dependencies and create the events from it from here 🤔 WDYT?

Reply:
I agree. It seems to belong more to `ZyncEvent` then to `ZyncWorker`.